### ISSUE TITLE
Gatling Maven plugin adds src/test/resources to the testing classpath

### ISSUE THREAD

**Issue by ViliusS** [gatling/gatling#4426]
Gatling should not bundle logback-test.xml file. If Gatling is used in Maven project it's very difficult to *correctly* override logging configuration if not impossible.

Maven picks up multiple XML configurations and logs are populated with things like:
```
08:26:35  05:26:35,336 |-INFO in ch.qos.logback.classic.LoggerContext[default] - Found resource [logback-test.xml] at [file:/srv/jenkins/workspace/load-testing_main/src/test/resources/logback-test.xml]
08:26:35  05:26:35,337 |-WARN in ch.qos.logback.classic.LoggerContext[default] - Resource [logback-test.xml] occurs multiple times on the classpath.
08:26:35  05:26:35,337 |-WARN in ch.qos.logback.classic.LoggerContext[default] - Resource [logback-test.xml] occurs at [file:/srv/jenkins/workspace/load-testing_main/target/test-classes/logback-test.xml]
08:26:35  05:26:35,337 |-WARN in ch.qos.logback.classic.LoggerContext[default] - Resource [logback-test.xml] occurs at [file:/srv/jenkins/workspace/load-testing_main/src/test/resources/logback-test.xml]
08:26:35  05:26:35,466 |-INFO in ch.qos.logback.classic.joran.action.ConfigurationAction - debug attribute not set
08:26:35  05:26:35,467 |-INFO in ch.qos.logback.core.joran.action.AppenderAction - About to instantiate appender of type [ch.qos.logback.core.ConsoleAppender]
08:26:35  05:26:35,473 |-INFO in ch.qos.logback.core.joran.action.AppenderAction - Naming appender as [CONSOLE]
08:26:35  05:26:35,482 |-INFO in ch.qos.logback.core.joran.action.NestedComplexPropertyIA - Assuming default type [ch.qos.logback.classic.encoder.PatternLayoutEncoder] for [encoder] property
08:26:35  05:26:35,534 |-INFO in ch.qos.logback.classic.joran.action.RootLoggerAction - Setting level of ROOT logger to WARN
08:26:35  05:26:35,534 |-INFO in ch.qos.logback.core.joran.action.AppenderRefAction - Attaching appender named [CONSOLE] to Logger[ROOT]
08:26:35  05:26:35,534 |-INFO in ch.qos.logback.classic.joran.action.ConfigurationAction - End of configuration.
08:26:35  05:26:35,535 |-INFO in ch.qos.logback.classic.joran.JoranConfigurator@25bbf683 - Registering current configuration as safe fallback point
```

It is a general best practice that libraries do not include any logging XML configuration but set it programmatically if needed. Or at least rename a file to something like logback-gatling.xml wrapped into `<included>` so it can be used as default and overridden if necessary (this is how Spring does it).

**Comment by slandelle** [comment]
Please check your logs properly:

Resource [logback-test.xml] occurs at [file:/srv/jenkins/workspace/load-testing_main/target/test-classes/logback-test.xml]
Resource [logback-test.xml] occurs at [file:/srv/jenkins/workspace/load-testing_main/src/test/resources/logback-test.xml

The logback-test.xml file is NOT bundled in Gatling binaries.
The issue happens because both resources and test-classes are in the classpath.
Please provide a standalone reproducer if you think the issue is in a Gatling component.

**Comment by ViliusS** [comment]
Thank you for your response. It looks like I was wrong. Garling indeed doesn't bundle logback-test.xml file.

However, there is a question then, why both src/test/resources and target/test-classes are on the testing classpath? I suppose it is a bug here? https://github.com/gatling/gatling-maven-plugin/blob/31c776c2cb3c24c2cb6ca1a71172dd91f051245b/src/main/java/io/gatling/mojo/AbstractGatlingMojo.java#L86

I don't follow the logic in gatling-maven-plugin but it looks like it adds resources folder to the test classpath depending on if gatling.conf exist or not. That folder can contain all sorts of resource files, including logback XML configuration, so such test classpath modification is wrong IMHO.

I just did a quick reproducible case with https://github.com/gatling/gatling-maven-plugin-demo-java

1. Clone the repo
2. Run `mvn clean gatling:test` . No issues with logback overloading.
3. Delete `src/test/resources/gatling.conf` which is basically empty commented out file.
4. Run `mvn clean gatling:test` again. This time you will get warnings about duplicate logback XML configuration.

**Comment by slandelle** [comment]
Fix released in 4.3.5.
